src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -51,182 +50,118 @@
-			if (SOUND_CATEGORIES_EXIST) {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					p.playSound(location, sound, category, volume, pitch);
-				} else {
-					p.playSound(location, soundEnum, category, volume, pitch);
-				}
-			} else {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					p.playSound(location, sound, volume, pitch);
-				} else {
-					p.playSound(location, soundEnum, volume, pitch);
-				}
-			}
-		}
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		if (locations != null) return "play sound " + sounds.toString(event, debug) +
+				(category != null ? " in " + category.toString(event, debug) : "") +
+				(volume != null ? " at volume " + volume.toString(event, debug) : "") +
+				(pitch != null ? " at pitch " + pitch.toString(event, debug) : "") +
+				(locations != null ? " at " + locations.toString(event, debug) : "") +
+				(players != null ? " for " + players.toString(event, debug) : "");
+		else return "play sound " + sounds.toString(event, debug) +
+				(volume != null ? " at volume " + volume.toString(event, debug) : "") +
+				(pitch != null ? " at pitch " + pitch.toString(event, debug) : "") +
+				(players != null ? " to " + players.toString(event, debug) : "");
 	}
-
-	private static void playSound(Location location, String[] sounds, SoundCategory category, float volume, float pitch) {
-		World w = location.getWorld();
-		for (String sound : sounds) {
-			Sound soundEnum = null;
-			try {
-				soundEnum = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
-			} catch (IllegalArgumentException ignored) {}
-			if (SOUND_CATEGORIES_EXIST) {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					w.playSound(location, sound, category, volume, pitch);
-				} else {
-					w.playSound(location, soundEnum, category, volume, pitch);
-				}
-			} else {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					w.playSound(location, sound, volume, pitch);
-				} else {
-					w.playSound(location, soundEnum, volume, pitch);
+	
+	@FunctionalInterface
+	private interface SoundReceiver<T, S> {
+		void play(@NotNull T receiver, @NotNull Location location, @NotNull S sound,
+	            @NotNull SoundCategory category, float volume, float pitch);
+		
+		static <T> void play(@NotNull SoundReceiver<T, String> stringReceiver,
+                @NotNull SoundReceiver<T, Sound> soundReceiver,
+                @NotNull T receiver, @NotNull Location location, @NotNull String[] sounds,
+                @NotNull SoundCategory category, float volume, float pitch) {
+			
```

Comment:
```suggestion
                @NotNull SoundCategory category, float volume, float pitch
		) {	
```
It seems some spaces have made it into the file in this area. Make sure all your indentations are tabs :)